frontend/src/components/SignUpForm/SignUpForm.jsx

```diff
@@ -21,19 +28,34 @@ export default function SignUpForm({ setUser }) {
             const response = await axios.post(
                 "http://localhost:3000/users/signup",
                 userData,
-                { withCredentials: true }
+                { withCredentials: true, validateStatus: () => true }
             );
 
-            setUsername("");
-            setPassword("");
-            setEmail("");
+            if (response.status === 200) {
+                setUsername("");
+                setPassword("");
+                setEmail("");
+
+                const user = response.data.user;
+                setUser(user);
 
-            const user = response.data.user;
-            setUser(user);
+                setSuccess(true);
 
-            alert("Sign Up Successful.");
+                setTimeout(() => {
+                    setSuccess(false);
+                    navigate("/");
+                }, 1000);
```

Comment:
Nice trick 👍

Reply:
The next UI improvement we can make is to add an animation like a spinner right before `const response = await axios.post(` and remove it right after.